Services/Queue/Processors/IndexImportProductsProcessor.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Wizzy\Search\Services\Queue\Processors;
+
+use Wizzy\Search\Services\Catalogue\ProductsManager;
+use Wizzy\Search\Services\Indexer\IndexerManager;
+use Wizzy\Search\Services\Indexer\IndexerOutput;
+use Wizzy\Search\Services\Store\StoreGeneralConfig;
+use Wizzy\Search\Services\Store\StoreManager;
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Filesystem;
+use Wizzy\Search\Model\Observer\ProductsObserver;
+
+class IndexImportProductsProcessor extends QueueProcessorBase
+{
+    private $productsIndexer;
+    private $storeGeneralConfig;
+    private $storeManager;
+    private $productsManager;
+    private $output;
+    private $filesystem;
+    private $productsObserver;
+
+    public function __construct(
+        IndexerManager $indexerManager,
+        ProductsManager $productsManager,
+        StoreManager $storeManager,
+        StoreGeneralConfig $storeGeneralConfig,
+        IndexerOutput $output,
+        Filesystem $filesystem,
+        ProductsObserver $productsObserver
+    ) {
+        $this->productsIndexer = $indexerManager->getProductsIndexer();
+        $this->storeGeneralConfig = $storeGeneralConfig;
+        $this->storeManager = $storeManager;
+        $this->productsManager = $productsManager;
+        $this->output = $output;
+        $this->filesystem = $filesystem;
+        $this->productsObserver = $productsObserver;
+    }
+
+    public function execute(array $data, $storeId)
+    {
+        $storeIds = $this->storeManager->getToSyncStoreIds($storeId);
```

Comment:
We don't need to loop through Store IDs here. $storeId we're already getting in the `execute` function. We need to execute for particular store only.